src/hooks/useClientScriptLoad.ts

```diff
@@ -0,0 +1,91 @@
+import { useState, useMemo, useEffect } from "react"
+import { isNostoLoaded } from "../components/helpers"
+import type { NostoClient } from "../types"
+import type { NostoProviderProps } from "../components/NostoProvider"
+
+type NostoScriptProps = Pick<NostoProviderProps, "account" | "host" | "shopifyMarkets" | "loadScript">
+
+export function useClientScriptLoad(props: NostoScriptProps) {
+		const { host, account, shopifyMarkets, loadScript = true } = props
+    const [ clientScriptLoadedState, setClientScriptLoadedState ] = useState(false)
+    const clientScriptLoaded = useMemo(() => clientScriptLoadedState, [clientScriptLoadedState])
+    
+		useEffect(() => { 
+			if (!window.nostojs) {
+				window.nostojs = (cb: (api: NostoClient) => void) => {
+					(window.nostojs.q = window.nostojs.q || []).push(cb)
+				}
+				window.nostojs(api => api.setAutoLoad(false))
+			}
+	
+			if (!isNostoLoaded() && !shopifyMarkets && loadScript) {
+				const script = document.createElement("script")
+				script.type = "text/javascript"
+				script.src = "//" + (host || "connect.nosto.com") + "/include/" + account
+				script.async = true
+				script.setAttribute("nosto-client-script", "")
+
+				script.onload = () => {
+				if (typeof "nostoReactTest" !== "undefined") {
```

Comment:
I would replace this line with the equivalent line in master
typeof "nostoReactTest" is "string"

Reply:
The `typeof "nostoReactTest" in window` seems sufficient for me